packages/postgres/src/test-utils.ts

```diff
@@ -0,0 +1,153 @@
+import pg from 'pg';
+
+import * as defaultPool from './default-pool';
+
+const POSTGRES_USER = 'postgres';
+const POSTGRES_HOST = 'localhost';
+const POSTGRES_DATABASE = 'postgres';
+
+export interface PostgresTestUtilsOptions {
+  user?: string;
+  host?: string;
+  defaultDatabase?: string;
+  database: string;
+  prepareAfterReset?: (client: pg.Client) => Promise<void>;
+}
+
+interface CreateDatabaseOptions {
+  dropExistingDatabase?: boolean;
+  database?: string;
+  templateDatabase?: string;
+  configurePool?: boolean;
+  prepare?: (client: pg.Client) => Promise<void>;
+}
+
+interface DropDatabaseOptions {
+  database?: string;
+  force?: boolean;
+  closePool?: boolean;
+}
+
+async function createDatabase(
+  options: PostgresTestUtilsOptions,
+  createOptions: CreateDatabaseOptions = {}
+): Promise<void> {
+  const dropExistingDatabase = createOptions.dropExistingDatabase ?? true;
+  const configurePool = createOptions.configurePool ?? true;
+
+  const client = new pg.Client({
+    ...getPoolConfig(options),
+    database: options.defaultDatabase ?? POSTGRES_DATABASE,
+  });
+  await client.connect();
+
+  const escapedDatabase = client.escapeIdentifier(
+    createOptions.database ?? getDatabaseNameForCurrentMochaWorker(options.database)
+  );
+  if (dropExistingDatabase ?? true) {
+    await client.query(`DROP DATABASE IF EXISTS ${escapedDatabase}`);
+  }
+
+  if (createOptions.templateDatabase) {
+    const escapedTemplateDatabase = client.escapeIdentifier(createOptions.templateDatabase);
+    await client.query(`CREATE DATABASE ${escapedDatabase} TEMPLATE ${escapedTemplateDatabase}`);
+  } else {
+    await client.query(`CREATE DATABASE ${escapedDatabase}`);
+  }
+
+  await client.end();
+
+  await createOptions.prepare?.(client);
+
+  if (configurePool) {
+    await defaultPool.initAsync(
+      {
+        user: options.user ?? POSTGRES_USER,
+        host: options.host ?? POSTGRES_HOST,
+        database: getDatabaseNameForCurrentMochaWorker(options.database),
+        // TODO: make these configurable?
```

Comment:
Seems like we may as well do it now, given that it's annoying to change a package and do the whole publish dance if we need to change this in the future?

Reply:
Yep, good call - I meant to do this before requesting review but forgot, sorry!